agent/consul/state/catalog.go

```diff
@@ -481,6 +481,24 @@ func (s *Store) ensureNodeTxn(tx *memdb.Txn, idx uint64, node *structs.Node) err
 	if err := tx.Insert("index", &IndexEntry{"nodes", idx}); err != nil {
 		return fmt.Errorf("failed updating index: %s", err)
 	}
+	// Update the node's service indexes as the node information is included
+	// in health queries and we would otherwise miss node updates in some cases
+	// for those queries.
+	services, err := tx.Get("services", "node", node.Node)
```

Comment:
You might use `updateAllServiceIndexesOfNode(tx *memdb.Txn, idx uint64, nodeID string)` instead that does this IIRC

Reply:
:+1: Did not see that func, fixed